inference-engine/src/inference_engine/ie_core.cpp

```diff
@@ -248,156 +247,77 @@ class Core::Impl : public ICore {
     std::map<std::string, PluginDescriptor> pluginRegistry;
     mutable std::mutex pluginsMutex;  // to lock parallel access to pluginRegistry and plugins
 
-    ExecutableNetwork LoadNetworkImpl(const std::string& modelPath, const std::string& deviceName,
-        const std::map<std::string, std::string>& config,
-        const RemoteContext::Ptr& context) {
-        THROW_IE_EXCEPTION_WITH_STATUS(NOT_IMPLEMENTED) << "mnosov: TBD";
-    }
-
-    ExecutableNetwork LoadNetworkImpl(const CNNNetwork& network, const std::string& deviceName,
-        const std::map<std::string, std::string>& config,
-        const RemoteContext::Ptr& context) {
-        //      NetworkCompilationContext context(network, compileConfig);
-        return DoLoadNetworkImpl(network, deviceName, config, context);
+    bool DeviceSupportsImportExport(const InferencePlugin& plugin) const {
+        bool supported = coreConfig.isModelCacheEnabled();
+        if (supported) {
+            std::vector<std::string> supportedMetricKeys = plugin.GetMetric(METRIC_KEY(SUPPORTED_METRICS), {});
+            auto it = std::find(supportedMetricKeys.begin(), supportedMetricKeys.end(),
+                                METRIC_KEY(IMPORT_EXPORT_SUPPORT));
+            supported = (it != supportedMetricKeys.end()) &&
+                        plugin.GetMetric(METRIC_KEY(IMPORT_EXPORT_SUPPORT), {});
+        }
+        return supported;
     }
 
-    ExecutableNetwork DoLoadNetworkImpl(const CNNNetwork& network, const std::string& deviceName,
-        const std::map<std::string, std::string>& configMap,
-        const RemoteContext::Ptr& context) {
-        OV_ITT_SCOPED_TASK(itt::domains::IE_LT, "Core::Impl::DoLoadNetworkImpl");
-
-        auto parsed = parseDeviceNameIntoConfig<std::string>(deviceName, configMap);
-        auto config = parsed._config;
-        std::string deviceFamily = parsed._deviceName; // MULTI:CPU -> MULTI, GPU.0 -> GPU, CPU -> CPU and so on
-
-        auto deviceSupportsImport = [&](ICore* core) -> bool {
-            OV_ITT_SCOPED_TASK(itt::domains::IE_LT, "Core::Impl::device_supports_import");
-            std::stringstream dummyStream;
-            bool supports = true;
+    ExecutableNetwork LoadNetworkImpl(const CNNNetwork& network,
+                                      InferencePlugin& plugin,
+                                      const std::map<std::string, std::string>& config,
+                                      const RemoteContext::Ptr& context,
+                                      const std::string& blobID) {
+        OV_ITT_SCOPED_TASK(itt::domains::IE_LT, "Core::Impl::LoadNetworkImpl");
+        ExecutableNetwork execNetwork;
+        execNetwork = context ? plugin.LoadNetwork(network, context, config) :
+                                plugin.LoadNetwork(network, config);
+        if (DeviceSupportsImportExport(plugin)) {
+            auto cacheManager = coreConfig.getCacheManager();
+            IE_ASSERT(cacheManager != nullptr);
```

Comment:
what if caching is not enabled? I suppose cache manager can be empty in this case.

Reply:
It is covered in DeviceSupportsImportExport. It will return 'false' if caching is not enabled. I moved it there to avoid duplication of "coreConfig.IsModelCachingEnabled() && DeviceSupportsImportExport()" in many places

Reply:
Ok, I see. Thanks